src/Datadog.Trace/Agent/NamedPipes/NamedPipesApiRequest.cs

```diff
@@ -0,0 +1,46 @@
+using System.IO.Pipes;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent.MessagePack;
+
+namespace Datadog.Trace.Agent.NamedPipes
+{
+    internal class NamedPipesApiRequest : IApiRequest
+    {
+        private readonly string _pipeName;
+        private readonly TraceRequest _traceRequest;
+
+        // `SpinLock` is a struct. A struct marked as `readonly` is copied each time a mutating function is called.
+        // When calling `_lock.Enter` and `_lock.Exit()` the `SpinLock` instance is copied. Calling `_lock.Exit()` raises an
+        // error as the instance does not hold the lock (System.Threading.SynchronizationLockException : The calling
+        // thread does not hold the lock.)
+        // For this reason, `_lock` is not marked as `readonly`
+        private SpinLock _lock = new SpinLock(enableThreadOwnerTracking: true);
```

Comment:
Doesn't look like we're using this lock anywhere.

Reply:
I will remove it if I don't use it. I had planned to.